core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1855,7 +1855,11 @@ object SnappySession extends Logging {
         evaluatePlan(df, session, key.sqlText, key)
       }
     }
-    CacheBuilder.newBuilder().maximumSize(300).build(loader)
+    val cacheSize = Property.PlanCacheSize.getOption(SnappyContext.initConf) match {
```

Comment:
>>Can't you use conf from SnappyContext.globalSparkContext and get rid if this separate initConf? 

We need initConf during SnappySession object's initialization. This initialization happens during the lead start and SparkContext.getOrCreate returns a null. 

>> I think if you have planCache as lazy, then it should work fine. 

The performance penalty of lazy is what I was trying to avoid. This conf business is clumsy.  

>> Alternatively you can just do manual testing for this property.

I just did that.